<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>06-Importación-de-datos-readr-readxl-httr-rvest-dbplyr</title>
    <meta charset="utf-8" />
    <meta name="author" content=" Joshua Kunst @jbkunst" />
    <meta name="date" content="2023-04-24" />
    <script src="libs/header-attrs-2.20/header-attrs.js"></script>
    <link rel="stylesheet" href="css/xaringan-themer.css" type="text/css" />
    <link rel="stylesheet" href="css/styles.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: left, middle, title-slide

.title[
# Computación Estadística EPG3308
]
.subtitle[
## 06 Importación de datos <code><small>readr readxl httr dbplyr</small></code>
]
.author[
### <br>Joshua Kunst<br><span class="citation">@jbkunst</span>
]
.date[
### 2023-04-24
]

---





## Orígenes de información

Los datos pueden venir de muchas fuentes distintas:
--


- Archivo de texto: csv, txt, tsv, fwf.
--

- Planillas, archivos excel.
--

- SPSS (?) SAS (?!) STATA (?!?!?).
--

- Bases de Datos. Esto es todo un mundo, existen muchos motores de bases de datos.
--

- Desde una página web: https://www.reclamos.cl/empresa/uber-eats
--

- Una API: https://climatologia.meteochile.gob.cl/application/productos/datosRecientesEma/330020/2022/05
--

- PDF? Documento?
--

- Algo más?
  
---
  
## `paquete::funcion` para cada tipo de origen

Origen            | `paquete::funcion`
------------------|---------
`xlsx`            |  `readxl::read_excel`
`txt`             |  `readr::read_delim` / `data.table::fread`
`csv`             |  `readr::read_csv` / `data.table::fread`
Bases de datos    |  Paquete [`dbplyr`](https://dbplyr.tidyverse.org/articles/dbplyr.html) `tbl(...)`
API               |  `httr::GET`
Página web        |  `rvest::read_html`
Archivo SPSS`sav` |  `haven::read_sav` / `haven::read_spss`
Archivo SAS       |  `haven::read_sas`
Archivo STATA     |  `haven::read_stata`
Archivo `pdf`     |  `pdftools::pdf_text` / `tabulizer::extract_tables`

---

## `read_csv` / `read_delim`

&lt;img src="images/readr/readr.png" width="70%" style="display: block; margin: auto;" /&gt;

---

## `read_csv` / `read_delim` parámetros

&lt;br/&gt;

&lt;img src="images/readr/readr-pars.png" width="70%" style="display: block; margin: auto;" /&gt;


---

count: false
 
## &lt;code&gt;readxl::read_excel&lt;/code&gt;
.panel1-read_excel-auto[

```r
*library(readxl)
```
]
 
.panel2-read_excel-auto[

]

---
count: false
 
## &lt;code&gt;readxl::read_excel&lt;/code&gt;
.panel1-read_excel-auto[

```r
library(readxl)

*ruta &lt;- here::here("data/muestrame_los_numeros.xlsx")
```
]
 
.panel2-read_excel-auto[

]

---
count: false
 
## &lt;code&gt;readxl::read_excel&lt;/code&gt;
.panel1-read_excel-auto[

```r
library(readxl)

ruta &lt;- here::here("data/muestrame_los_numeros.xlsx")
*ruta
```
]
 
.panel2-read_excel-auto[

```
## [1] "D:/git/computacion-estadistica-EPG3308/data/muestrame_los_numeros.xlsx"
```
]

---
count: false
 
## &lt;code&gt;readxl::read_excel&lt;/code&gt;
.panel1-read_excel-auto[

```r
library(readxl)

ruta &lt;- here::here("data/muestrame_los_numeros.xlsx")
ruta

*read_excel(ruta)
```
]
 
.panel2-read_excel-auto[

```
## [1] "D:/git/computacion-estadistica-EPG3308/data/muestrame_los_numeros.xlsx"
```

```
## # A tibble: 16 × 3
##    region cuarto valor
##    &lt;chr&gt;  &lt;chr&gt;  &lt;dbl&gt;
##  1 sur    Q1       100
##  2 sur    Q2       150
##  3 sur    Q3       225
##  4 sur    Q4       290
##  5 norte  Q1       150
##  6 norte  Q2       160
##  7 norte  Q3       180
##  8 norte  Q4       300
##  9 este   Q1       180
## 10 este   Q2       200
## 11 este   Q3       200
## 12 este   Q4       240
## 13 oeste  Q1       250
## 14 oeste  Q2       250
## 15 oeste  Q3       300
## 16 oeste  Q4       170
```
]

&lt;style&gt;
.panel1-read_excel-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel2-read_excel-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel3-read_excel-auto {
  color: black;
  width: NA%;
  hight: 33%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
&lt;/style&gt;





---

count: false
 
## &lt;code&gt;readr::read_csv&lt;/code&gt;
.panel1-read_csv-auto[

```r
*library(readr)
```
]
 
.panel2-read_csv-auto[

]

---
count: false
 
## &lt;code&gt;readr::read_csv&lt;/code&gt;
.panel1-read_csv-auto[

```r
library(readr)

*ruta &lt;- here::here("data/pollution.csv")
```
]
 
.panel2-read_csv-auto[

]

---
count: false
 
## &lt;code&gt;readr::read_csv&lt;/code&gt;
.panel1-read_csv-auto[

```r
library(readr)

ruta &lt;- here::here("data/pollution.csv")
*ruta
```
]
 
.panel2-read_csv-auto[

```
## [1] "D:/git/computacion-estadistica-EPG3308/data/pollution.csv"
```
]

---
count: false
 
## &lt;code&gt;readr::read_csv&lt;/code&gt;
.panel1-read_csv-auto[

```r
library(readr)

ruta &lt;- here::here("data/pollution.csv")
ruta

*read_csv(ruta)
```
]
 
.panel2-read_csv-auto[

```
## [1] "D:/git/computacion-estadistica-EPG3308/data/pollution.csv"
```

```
## # A tibble: 6 × 3
##   city     size  amount
##   &lt;chr&gt;    &lt;chr&gt;  &lt;dbl&gt;
## 1 New York large     23
## 2 New York small     14
## 3 London   large     22
## 4 London   small     16
## 5 Beijing  large    121
## 6 Beijing  small     56
```
]

&lt;style&gt;
.panel1-read_csv-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel2-read_csv-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel3-read_csv-auto {
  color: black;
  width: NA%;
  hight: 33%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
&lt;/style&gt;





---

count: false
 
## &lt;code&gt;httr::GET&lt;/code&gt;
.panel1-GET-auto[

```r
*library(httr)
```
]
 
.panel2-GET-auto[

]

---
count: false
 
## &lt;code&gt;httr::GET&lt;/code&gt;
.panel1-GET-auto[

```r
library(httr)
*library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
```
]
 
.panel2-GET-auto[

]

---
count: false
 
## &lt;code&gt;httr::GET&lt;/code&gt;
.panel1-GET-auto[

```r
library(httr)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc

*url &lt;- "https://mindicador.cl/api/tasa_desempleo/"
```
]
 
.panel2-GET-auto[

]

---
count: false
 
## &lt;code&gt;httr::GET&lt;/code&gt;
.panel1-GET-auto[

```r
library(httr)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc

url &lt;- "https://mindicador.cl/api/tasa_desempleo/"

*content(httr::GET(url))
```
]
 
.panel2-GET-auto[

```
## $version
## [1] "1.7.0"
## 
## $autor
## [1] "mindicador.cl"
## 
## $codigo
## [1] "tasa_desempleo"
## 
## $nombre
## [1] "Tasa de desempleo"
## 
## $unidad_medida
## [1] "Porcentaje"
## 
## $serie
## $serie[[1]]
## $serie[[1]]$fecha
## [1] "2023-02-01T03:00:00.000Z"
## 
## $serie[[1]]$valor
## [1] 8.37
## 
## 
## $serie[[2]]
## $serie[[2]]$fecha
## [1] "2023-01-01T03:00:00.000Z"
## 
## $serie[[2]]$valor
## [1] 8.04
## 
## 
## $serie[[3]]
## $serie[[3]]$fecha
## [1] "2022-12-01T03:00:00.000Z"
## 
## $serie[[3]]$valor
## [1] 7.86
## 
## 
## $serie[[4]]
## $serie[[4]]$fecha
## [1] "2022-11-01T03:00:00.000Z"
## 
## $serie[[4]]$valor
## [1] 7.95
## 
## 
## $serie[[5]]
## $serie[[5]]$fecha
## [1] "2022-10-01T03:00:00.000Z"
## 
## $serie[[5]]$valor
## [1] 7.97
## 
## 
## $serie[[6]]
## $serie[[6]]$fecha
## [1] "2022-09-01T04:00:00.000Z"
## 
## $serie[[6]]$valor
## [1] 8.04
## 
## 
## $serie[[7]]
## $serie[[7]]$fecha
## [1] "2022-08-01T04:00:00.000Z"
## 
## $serie[[7]]$valor
## [1] 7.93
## 
## 
## $serie[[8]]
## $serie[[8]]$fecha
## [1] "2022-07-01T04:00:00.000Z"
## 
## $serie[[8]]$valor
## [1] 7.92
## 
## 
## $serie[[9]]
## $serie[[9]]$fecha
## [1] "2022-06-01T04:00:00.000Z"
## 
## $serie[[9]]$valor
## [1] 7.81
## 
## 
## $serie[[10]]
## $serie[[10]]$fecha
## [1] "2022-05-01T04:00:00.000Z"
## 
## $serie[[10]]$valor
## [1] 7.8
## 
## 
## $serie[[11]]
## $serie[[11]]$fecha
## [1] "2022-04-01T03:00:00.000Z"
## 
## $serie[[11]]$valor
## [1] 7.75
## 
## 
## $serie[[12]]
## $serie[[12]]$fecha
## [1] "2022-03-01T03:00:00.000Z"
## 
## $serie[[12]]$valor
## [1] 7.8
## 
## 
## $serie[[13]]
## $serie[[13]]$fecha
## [1] "2022-02-01T03:00:00.000Z"
## 
## $serie[[13]]$valor
## [1] 7.5
## 
## 
## $serie[[14]]
## $serie[[14]]$fecha
## [1] "2022-01-01T03:00:00.000Z"
## 
## $serie[[14]]$valor
## [1] 7.27
## 
## 
## $serie[[15]]
## $serie[[15]]$fecha
## [1] "2021-12-01T03:00:00.000Z"
## 
## $serie[[15]]$valor
## [1] 7.19
## 
## 
## $serie[[16]]
## $serie[[16]]$fecha
## [1] "2021-11-01T03:00:00.000Z"
## 
## $serie[[16]]$valor
## [1] 7.53
## 
## 
## $serie[[17]]
## $serie[[17]]$fecha
## [1] "2021-10-01T03:00:00.000Z"
## 
## $serie[[17]]$valor
## [1] 8.08
## 
## 
## $serie[[18]]
## $serie[[18]]$fecha
## [1] "2021-09-01T04:00:00.000Z"
## 
## $serie[[18]]$valor
## [1] 8.41
## 
## 
## $serie[[19]]
## $serie[[19]]$fecha
## [1] "2021-08-01T04:00:00.000Z"
## 
## $serie[[19]]$valor
## [1] 8.55
## 
## 
## $serie[[20]]
## $serie[[20]]$fecha
## [1] "2021-07-01T04:00:00.000Z"
## 
## $serie[[20]]$valor
## [1] 8.93
## 
## 
## $serie[[21]]
## $serie[[21]]$fecha
## [1] "2021-06-01T04:00:00.000Z"
## 
## $serie[[21]]$valor
## [1] 9.5
## 
## 
## $serie[[22]]
## $serie[[22]]$fecha
## [1] "2021-05-01T04:00:00.000Z"
## 
## $serie[[22]]$valor
## [1] 10.04
## 
## 
## $serie[[23]]
## $serie[[23]]$fecha
## [1] "2021-04-01T03:00:00.000Z"
## 
## $serie[[23]]$valor
## [1] 10.24
## 
## 
## $serie[[24]]
## $serie[[24]]$fecha
## [1] "2021-03-01T03:00:00.000Z"
## 
## $serie[[24]]$valor
## [1] 10.35
## 
## 
## $serie[[25]]
## $serie[[25]]$fecha
## [1] "2021-02-01T03:00:00.000Z"
## 
## $serie[[25]]$valor
## [1] 10.3
## 
## 
## $serie[[26]]
## $serie[[26]]$fecha
## [1] "2021-01-01T03:00:00.000Z"
## 
## $serie[[26]]$valor
## [1] 10.23
## 
## 
## $serie[[27]]
## $serie[[27]]$fecha
## [1] "2020-12-01T03:00:00.000Z"
## 
## $serie[[27]]$valor
## [1] 10.29
## 
## 
## $serie[[28]]
## $serie[[28]]$fecha
## [1] "2020-11-01T03:00:00.000Z"
## 
## $serie[[28]]$valor
## [1] 10.76
## 
## 
## $serie[[29]]
## $serie[[29]]$fecha
## [1] "2020-10-01T03:00:00.000Z"
## 
## $serie[[29]]$valor
## [1] 11.58
## 
## 
## $serie[[30]]
## $serie[[30]]$fecha
## [1] "2020-09-01T04:00:00.000Z"
## 
## $serie[[30]]$valor
## [1] 12.35
## 
## 
## $serie[[31]]
## $serie[[31]]$fecha
## [1] "2020-08-01T04:00:00.000Z"
## 
## $serie[[31]]$valor
## [1] 12.93
```
]

---
count: false
 
## &lt;code&gt;httr::GET&lt;/code&gt;
.panel1-GET-auto[

```r
library(httr)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc

url &lt;- "https://mindicador.cl/api/tasa_desempleo/"

content(httr::GET(url)) |&gt;
* pluck("serie")
```
]
 
.panel2-GET-auto[

```
## [[1]]
## [[1]]$fecha
## [1] "2023-02-01T03:00:00.000Z"
## 
## [[1]]$valor
## [1] 8.37
## 
## 
## [[2]]
## [[2]]$fecha
## [1] "2023-01-01T03:00:00.000Z"
## 
## [[2]]$valor
## [1] 8.04
## 
## 
## [[3]]
## [[3]]$fecha
## [1] "2022-12-01T03:00:00.000Z"
## 
## [[3]]$valor
## [1] 7.86
## 
## 
## [[4]]
## [[4]]$fecha
## [1] "2022-11-01T03:00:00.000Z"
## 
## [[4]]$valor
## [1] 7.95
## 
## 
## [[5]]
## [[5]]$fecha
## [1] "2022-10-01T03:00:00.000Z"
## 
## [[5]]$valor
## [1] 7.97
## 
## 
## [[6]]
## [[6]]$fecha
## [1] "2022-09-01T04:00:00.000Z"
## 
## [[6]]$valor
## [1] 8.04
## 
## 
## [[7]]
## [[7]]$fecha
## [1] "2022-08-01T04:00:00.000Z"
## 
## [[7]]$valor
## [1] 7.93
## 
## 
## [[8]]
## [[8]]$fecha
## [1] "2022-07-01T04:00:00.000Z"
## 
## [[8]]$valor
## [1] 7.92
## 
## 
## [[9]]
## [[9]]$fecha
## [1] "2022-06-01T04:00:00.000Z"
## 
## [[9]]$valor
## [1] 7.81
## 
## 
## [[10]]
## [[10]]$fecha
## [1] "2022-05-01T04:00:00.000Z"
## 
## [[10]]$valor
## [1] 7.8
## 
## 
## [[11]]
## [[11]]$fecha
## [1] "2022-04-01T03:00:00.000Z"
## 
## [[11]]$valor
## [1] 7.75
## 
## 
## [[12]]
## [[12]]$fecha
## [1] "2022-03-01T03:00:00.000Z"
## 
## [[12]]$valor
## [1] 7.8
## 
## 
## [[13]]
## [[13]]$fecha
## [1] "2022-02-01T03:00:00.000Z"
## 
## [[13]]$valor
## [1] 7.5
## 
## 
## [[14]]
## [[14]]$fecha
## [1] "2022-01-01T03:00:00.000Z"
## 
## [[14]]$valor
## [1] 7.27
## 
## 
## [[15]]
## [[15]]$fecha
## [1] "2021-12-01T03:00:00.000Z"
## 
## [[15]]$valor
## [1] 7.19
## 
## 
## [[16]]
## [[16]]$fecha
## [1] "2021-11-01T03:00:00.000Z"
## 
## [[16]]$valor
## [1] 7.53
## 
## 
## [[17]]
## [[17]]$fecha
## [1] "2021-10-01T03:00:00.000Z"
## 
## [[17]]$valor
## [1] 8.08
## 
## 
## [[18]]
## [[18]]$fecha
## [1] "2021-09-01T04:00:00.000Z"
## 
## [[18]]$valor
## [1] 8.41
## 
## 
## [[19]]
## [[19]]$fecha
## [1] "2021-08-01T04:00:00.000Z"
## 
## [[19]]$valor
## [1] 8.55
## 
## 
## [[20]]
## [[20]]$fecha
## [1] "2021-07-01T04:00:00.000Z"
## 
## [[20]]$valor
## [1] 8.93
## 
## 
## [[21]]
## [[21]]$fecha
## [1] "2021-06-01T04:00:00.000Z"
## 
## [[21]]$valor
## [1] 9.5
## 
## 
## [[22]]
## [[22]]$fecha
## [1] "2021-05-01T04:00:00.000Z"
## 
## [[22]]$valor
## [1] 10.04
## 
## 
## [[23]]
## [[23]]$fecha
## [1] "2021-04-01T03:00:00.000Z"
## 
## [[23]]$valor
## [1] 10.24
## 
## 
## [[24]]
## [[24]]$fecha
## [1] "2021-03-01T03:00:00.000Z"
## 
## [[24]]$valor
## [1] 10.35
## 
## 
## [[25]]
## [[25]]$fecha
## [1] "2021-02-01T03:00:00.000Z"
## 
## [[25]]$valor
## [1] 10.3
## 
## 
## [[26]]
## [[26]]$fecha
## [1] "2021-01-01T03:00:00.000Z"
## 
## [[26]]$valor
## [1] 10.23
## 
## 
## [[27]]
## [[27]]$fecha
## [1] "2020-12-01T03:00:00.000Z"
## 
## [[27]]$valor
## [1] 10.29
## 
## 
## [[28]]
## [[28]]$fecha
## [1] "2020-11-01T03:00:00.000Z"
## 
## [[28]]$valor
## [1] 10.76
## 
## 
## [[29]]
## [[29]]$fecha
## [1] "2020-10-01T03:00:00.000Z"
## 
## [[29]]$valor
## [1] 11.58
## 
## 
## [[30]]
## [[30]]$fecha
## [1] "2020-09-01T04:00:00.000Z"
## 
## [[30]]$valor
## [1] 12.35
## 
## 
## [[31]]
## [[31]]$fecha
## [1] "2020-08-01T04:00:00.000Z"
## 
## [[31]]$valor
## [1] 12.93
```
]

---
count: false
 
## &lt;code&gt;httr::GET&lt;/code&gt;
.panel1-GET-auto[

```r
library(httr)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc

url &lt;- "https://mindicador.cl/api/tasa_desempleo/"

content(httr::GET(url)) |&gt;
  pluck("serie") |&gt;
* purrr::map_df(tibble::as_tibble)
```
]
 
.panel2-GET-auto[

```
## # A tibble: 31 × 2
##    fecha                    valor
##    &lt;chr&gt;                    &lt;dbl&gt;
##  1 2023-02-01T03:00:00.000Z  8.37
##  2 2023-01-01T03:00:00.000Z  8.04
##  3 2022-12-01T03:00:00.000Z  7.86
##  4 2022-11-01T03:00:00.000Z  7.95
##  5 2022-10-01T03:00:00.000Z  7.97
##  6 2022-09-01T04:00:00.000Z  8.04
##  7 2022-08-01T04:00:00.000Z  7.93
##  8 2022-07-01T04:00:00.000Z  7.92
##  9 2022-06-01T04:00:00.000Z  7.81
## 10 2022-05-01T04:00:00.000Z  7.8 
## # … with 21 more rows
```
]

---
count: false
 
## &lt;code&gt;httr::GET&lt;/code&gt;
.panel1-GET-auto[

```r
library(httr)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc

url &lt;- "https://mindicador.cl/api/tasa_desempleo/"

content(httr::GET(url)) |&gt;
  pluck("serie") |&gt;
  purrr::map_df(tibble::as_tibble) |&gt;
* mutate(fecha = ymd_hms(fecha))
```
]
 
.panel2-GET-auto[

```
## # A tibble: 31 × 2
##    fecha               valor
##    &lt;dttm&gt;              &lt;dbl&gt;
##  1 2023-02-01 03:00:00  8.37
##  2 2023-01-01 03:00:00  8.04
##  3 2022-12-01 03:00:00  7.86
##  4 2022-11-01 03:00:00  7.95
##  5 2022-10-01 03:00:00  7.97
##  6 2022-09-01 04:00:00  8.04
##  7 2022-08-01 04:00:00  7.93
##  8 2022-07-01 04:00:00  7.92
##  9 2022-06-01 04:00:00  7.81
## 10 2022-05-01 04:00:00  7.8 
## # … with 21 more rows
```
]

---
count: false
 
## &lt;code&gt;httr::GET&lt;/code&gt;
.panel1-GET-auto[

```r
library(httr)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc

url &lt;- "https://mindicador.cl/api/tasa_desempleo/"

content(httr::GET(url)) |&gt;
  pluck("serie") |&gt;
  purrr::map_df(tibble::as_tibble) |&gt;
  mutate(fecha = ymd_hms(fecha)) |&gt;
* ggplot(aes(fecha, valor))
```
]
 
.panel2-GET-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/GET_auto_08_output-1.svg" width="100%" /&gt;
]

---
count: false
 
## &lt;code&gt;httr::GET&lt;/code&gt;
.panel1-GET-auto[

```r
library(httr)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc

url &lt;- "https://mindicador.cl/api/tasa_desempleo/"

content(httr::GET(url)) |&gt;
  pluck("serie") |&gt;
  purrr::map_df(tibble::as_tibble) |&gt;
  mutate(fecha = ymd_hms(fecha)) |&gt;
  ggplot(aes(fecha, valor)) +
* geom_line(size = 1.2, color = "steelblue")
```
]
 
.panel2-GET-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/GET_auto_09_output-1.svg" width="100%" /&gt;
]

---
count: false
 
## &lt;code&gt;httr::GET&lt;/code&gt;
.panel1-GET-auto[

```r
library(httr)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc

url &lt;- "https://mindicador.cl/api/tasa_desempleo/"

content(httr::GET(url)) |&gt;
  pluck("serie") |&gt;
  purrr::map_df(tibble::as_tibble) |&gt;
  mutate(fecha = ymd_hms(fecha)) |&gt;
  ggplot(aes(fecha, valor)) +
  geom_line(size = 1.2, color = "steelblue") +
* ylim(c(0, NA))
```
]
 
.panel2-GET-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/GET_auto_10_output-1.svg" width="100%" /&gt;
]

---
count: false
 
## &lt;code&gt;httr::GET&lt;/code&gt;
.panel1-GET-auto[

```r
library(httr)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc

url &lt;- "https://mindicador.cl/api/tasa_desempleo/"

content(httr::GET(url)) |&gt;
  pluck("serie") |&gt;
  purrr::map_df(tibble::as_tibble) |&gt;
  mutate(fecha = ymd_hms(fecha)) |&gt;
  ggplot(aes(fecha, valor)) +
  geom_line(size = 1.2, color = "steelblue") +
  ylim(c(0, NA)) +
* labs(title = "Tasa de desempleo")
```
]
 
.panel2-GET-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/GET_auto_11_output-1.svg" width="100%" /&gt;
]

&lt;style&gt;
.panel1-GET-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel2-GET-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel3-GET-auto {
  color: black;
  width: NA%;
  hight: 33%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
&lt;/style&gt;





---

count: false
 
## Bases de datos &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr-auto[

```r
*library(dplyr)
```
]
 
.panel2-dbplyr-auto[

]

---
count: false
 
## Bases de datos &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr-auto[

```r
library(dplyr)
*library(dbplyr)
```
]
 
.panel2-dbplyr-auto[

]

---
count: false
 
## Bases de datos &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr-auto[

```r
library(dplyr)
library(dbplyr)
*library(DBI)
```
]
 
.panel2-dbplyr-auto[

]

---
count: false
 
## Bases de datos &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr-auto[

```r
library(dplyr)
library(dbplyr)
library(DBI)
*library(RMariaDB)
```
]
 
.panel2-dbplyr-auto[

]

---
count: false
 
## Bases de datos &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr-auto[

```r
library(dplyr)
library(dbplyr)
library(DBI)
library(RMariaDB)

# https://docs.rfam.org/en/latest/database.html
*con &lt;-  dbConnect(
* RMariaDB::MariaDB(),
* host = "mysql-rfam-public.ebi.ac.uk",
* db = "Rfam",
* user = "rfamro",
* port = 4497
* )
```
]
 
.panel2-dbplyr-auto[

]

---
count: false
 
## Bases de datos &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr-auto[

```r
library(dplyr)
library(dbplyr)
library(DBI)
library(RMariaDB)

# https://docs.rfam.org/en/latest/database.html
con &lt;-  dbConnect(
  RMariaDB::MariaDB(),
  host = "mysql-rfam-public.ebi.ac.uk",
  db = "Rfam",
  user = "rfamro",
  port = 4497
  )

*tbl(con, "taxonomy")
```
]
 
.panel2-dbplyr-auto[

```
## # Source:   table&lt;taxonomy&gt; [?? x 5]
## # Database: mysql  [rfamro@mysql-rfam-public.ebi.ac.uk:NA/Rfam]
##    ncbi_id species                      tax_string               tree_…¹ align…²
##      &lt;int&gt; &lt;chr&gt;                        &lt;chr&gt;                    &lt;chr&gt;   &lt;chr&gt;  
##  1       7 Azorhizobium caulinodans     Bacteria; Proteobacteri… Azorhi… Azorhi…
##  2       9 Buchnera aphidicola          Bacteria; Proteobacteri… Buchne… Buchne…
##  3      11 [Cellvibrio] gilvus          Bacteria; Actinobacteri… [Cellv… [Cellv…
##  4      14 Dictyoglomus thermophilum    Bacteria; Dictyoglomi; … Dictyo… Dictyo…
##  5      17 Methylophilus methylotrophus Bacteria; Proteobacteri… Methyl… Methyl…
##  6      19 Pelobacter carbinolicus      Bacteria; Proteobacteri… Peloba… Peloba…
##  7      21 Phenylobacterium immobile    Bacteria; Proteobacteri… Phenyl… Phenyl…
##  8      23 Shewanella colwelliana       Bacteria; Proteobacteri… Shewan… Shewan…
##  9      24 Shewanella putrefaciens      Bacteria; Proteobacteri… Shewan… Shewan…
## 10      25 Shewanella hanedai           Bacteria; Proteobacteri… Shewan… Shewan…
## # … with more rows, and abbreviated variable names ¹​tree_display_name,
## #   ²​align_display_name
```
]

---
count: false
 
## Bases de datos &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr-auto[

```r
library(dplyr)
library(dbplyr)
library(DBI)
library(RMariaDB)

# https://docs.rfam.org/en/latest/database.html
con &lt;-  dbConnect(
  RMariaDB::MariaDB(),
  host = "mysql-rfam-public.ebi.ac.uk",
  db = "Rfam",
  user = "rfamro",
  port = 4497
  )

tbl(con, "taxonomy") |&gt;
* select(ncbi_id, species, tax_string)
```
]
 
.panel2-dbplyr-auto[

```
## # Source:   SQL [?? x 3]
## # Database: mysql  [rfamro@mysql-rfam-public.ebi.ac.uk:NA/Rfam]
##    ncbi_id species                      tax_string                              
##      &lt;int&gt; &lt;chr&gt;                        &lt;chr&gt;                                   
##  1       7 Azorhizobium caulinodans     Bacteria; Proteobacteria; Alphaproteoba…
##  2       9 Buchnera aphidicola          Bacteria; Proteobacteria; Gammaproteoba…
##  3      11 [Cellvibrio] gilvus          Bacteria; Actinobacteria; Actinobacteri…
##  4      14 Dictyoglomus thermophilum    Bacteria; Dictyoglomi; Dictyoglomales; …
##  5      17 Methylophilus methylotrophus Bacteria; Proteobacteria; Betaproteobac…
##  6      19 Pelobacter carbinolicus      Bacteria; Proteobacteria; Deltaproteoba…
##  7      21 Phenylobacterium immobile    Bacteria; Proteobacteria; Alphaproteoba…
##  8      23 Shewanella colwelliana       Bacteria; Proteobacteria; Gammaproteoba…
##  9      24 Shewanella putrefaciens      Bacteria; Proteobacteria; Gammaproteoba…
## 10      25 Shewanella hanedai           Bacteria; Proteobacteria; Gammaproteoba…
## # … with more rows
```
]

---
count: false
 
## Bases de datos &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr-auto[

```r
library(dplyr)
library(dbplyr)
library(DBI)
library(RMariaDB)

# https://docs.rfam.org/en/latest/database.html
con &lt;-  dbConnect(
  RMariaDB::MariaDB(),
  host = "mysql-rfam-public.ebi.ac.uk",
  db = "Rfam",
  user = "rfamro",
  port = 4497
  )

tbl(con, "taxonomy") |&gt;
  select(ncbi_id, species, tax_string) |&gt;
* filter(ncbi_id == 10116)
```
]
 
.panel2-dbplyr-auto[

```
## # Source:   SQL [1 x 3]
## # Database: mysql  [rfamro@mysql-rfam-public.ebi.ac.uk:NA/Rfam]
##   ncbi_id species                        tax_string                             
##     &lt;int&gt; &lt;chr&gt;                          &lt;chr&gt;                                  
## 1   10116 Rattus norvegicus (Norway rat) Eukaryota; Metazoa; Chordata; Craniata…
```
]

---
count: false
 
## Bases de datos &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr-auto[

```r
library(dplyr)
library(dbplyr)
library(DBI)
library(RMariaDB)

# https://docs.rfam.org/en/latest/database.html
con &lt;-  dbConnect(
  RMariaDB::MariaDB(),
  host = "mysql-rfam-public.ebi.ac.uk",
  db = "Rfam",
  user = "rfamro",
  port = 4497
  )

tbl(con, "taxonomy") |&gt;
  select(ncbi_id, species, tax_string) |&gt;
  filter(ncbi_id == 10116) |&gt;
* show_query()
```
]
 
.panel2-dbplyr-auto[

```
## &lt;SQL&gt;
## SELECT `ncbi_id`, `species`, `tax_string`
## FROM `taxonomy`
## WHERE (`ncbi_id` = 10116.0)
```
]

---
count: false
 
## Bases de datos &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr-auto[

```r
library(dplyr)
library(dbplyr)
library(DBI)
library(RMariaDB)

# https://docs.rfam.org/en/latest/database.html
con &lt;-  dbConnect(
  RMariaDB::MariaDB(),
  host = "mysql-rfam-public.ebi.ac.uk",
  db = "Rfam",
  user = "rfamro",
  port = 4497
  )

tbl(con, "taxonomy") |&gt;
  select(ncbi_id, species, tax_string) |&gt;
  filter(ncbi_id == 10116) |&gt;
  show_query() |&gt;
* collect()
```
]
 
.panel2-dbplyr-auto[

```
## &lt;SQL&gt;
## SELECT `ncbi_id`, `species`, `tax_string`
## FROM `taxonomy`
## WHERE (`ncbi_id` = 10116.0)
```

```
## # A tibble: 1 × 3
##   ncbi_id species                        tax_string                             
##     &lt;int&gt; &lt;chr&gt;                          &lt;chr&gt;                                  
## 1   10116 Rattus norvegicus (Norway rat) Eukaryota; Metazoa; Chordata; Craniata…
```
]

&lt;style&gt;
.panel1-dbplyr-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel2-dbplyr-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel3-dbplyr-auto {
  color: black;
  width: NA%;
  hight: 33%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
&lt;/style&gt;






---

count: false
 
## Bases de datos (cont.) &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr_cont-auto[

```r
*con
```
]
 
.panel2-dbplyr_cont-auto[

```
## &lt;MariaDBConnection&gt;
##   Host:    mysql-rfam-public.ebi.ac.uk
##   Server:  
##   Client:
```
]

---
count: false
 
## Bases de datos (cont.) &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr_cont-auto[

```r
con

*DBI::dbListTables(con)
```
]
 
.panel2-dbplyr_cont-auto[

```
## &lt;MariaDBConnection&gt;
##   Host:    mysql-rfam-public.ebi.ac.uk
##   Server:  
##   Client:
```

```
##  [1] "_annotated_file"             "_family_file"               
##  [3] "_genome_data"                "_lock"                      
##  [5] "_overlap"                    "_overlap_membership"        
##  [7] "_post_process"               "alignment_and_tree"         
##  [9] "author"                      "clan"                       
## [11] "clan_database_link"          "clan_literature_reference"  
## [13] "clan_membership"             "database_link"              
## [15] "db_version"                  "dead_clan"                  
## [17] "dead_family"                 "ensembl_names"              
## [19] "family"                      "family_author"              
## [21] "family_literature_reference" "family_long"                
## [23] "family_ncbi"                 "features"                   
## [25] "full_region"                 "genome"                     
## [27] "genome_temp"                 "genseq"                     
## [29] "genseq_temp"                 "html_alignment"             
## [31] "keywords"                    "literature_reference"       
## [33] "matches_and_fasta"           "motif"                      
## [35] "motif_database_link"         "motif_family_stats"         
## [37] "motif_file"                  "motif_literature"           
## [39] "motif_matches"               "motif_old"                  
## [41] "motif_pdb"                   "motif_ss_image"             
## [43] "pdb"                         "pdb_full_region"            
## [45] "pdb_full_region_old"         "pdb_rfam_reg"               
## [47] "pdb_sequence"                "processed_data"             
## [49] "pseudoknot"                  "refseq"                     
## [51] "refseq_full_region"          "rfamseq"                    
## [53] "rfamseq_temp"                "rnacentral_matches"         
## [55] "rscape_annotations"          "secondary_structure_image"  
## [57] "seed_region"                 "sunburst"                   
## [59] "taxonomic_tree"              "taxonomy"                   
## [61] "taxonomy_websearch"          "version"                    
## [63] "wikitext"
```
]

---
count: false
 
## Bases de datos (cont.) &lt;code&gt;dbplyr&lt;/code&gt;
.panel1-dbplyr_cont-auto[

```r
con

DBI::dbListTables(con)

*DBI::dbListFields(con, "taxonomy")
```
]
 
.panel2-dbplyr_cont-auto[

```
## &lt;MariaDBConnection&gt;
##   Host:    mysql-rfam-public.ebi.ac.uk
##   Server:  
##   Client:
```

```
##  [1] "_annotated_file"             "_family_file"               
##  [3] "_genome_data"                "_lock"                      
##  [5] "_overlap"                    "_overlap_membership"        
##  [7] "_post_process"               "alignment_and_tree"         
##  [9] "author"                      "clan"                       
## [11] "clan_database_link"          "clan_literature_reference"  
## [13] "clan_membership"             "database_link"              
## [15] "db_version"                  "dead_clan"                  
## [17] "dead_family"                 "ensembl_names"              
## [19] "family"                      "family_author"              
## [21] "family_literature_reference" "family_long"                
## [23] "family_ncbi"                 "features"                   
## [25] "full_region"                 "genome"                     
## [27] "genome_temp"                 "genseq"                     
## [29] "genseq_temp"                 "html_alignment"             
## [31] "keywords"                    "literature_reference"       
## [33] "matches_and_fasta"           "motif"                      
## [35] "motif_database_link"         "motif_family_stats"         
## [37] "motif_file"                  "motif_literature"           
## [39] "motif_matches"               "motif_old"                  
## [41] "motif_pdb"                   "motif_ss_image"             
## [43] "pdb"                         "pdb_full_region"            
## [45] "pdb_full_region_old"         "pdb_rfam_reg"               
## [47] "pdb_sequence"                "processed_data"             
## [49] "pseudoknot"                  "refseq"                     
## [51] "refseq_full_region"          "rfamseq"                    
## [53] "rfamseq_temp"                "rnacentral_matches"         
## [55] "rscape_annotations"          "secondary_structure_image"  
## [57] "seed_region"                 "sunburst"                   
## [59] "taxonomic_tree"              "taxonomy"                   
## [61] "taxonomy_websearch"          "version"                    
## [63] "wikitext"
```

```
## [1] "ncbi_id"            "species"            "tax_string"        
## [4] "tree_display_name"  "align_display_name"
```
]

&lt;style&gt;
.panel1-dbplyr_cont-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel2-dbplyr_cont-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel3-dbplyr_cont-auto {
  color: black;
  width: NA%;
  hight: 33%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
&lt;/style&gt;





---

count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
*library(rvest)
```
]
 
.panel2-rvest-auto[

]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
*library(lubridate)
```
]
 
.panel2-rvest-auto[

]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

*url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"
```
]
 
.panel2-rvest-auto[

]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"

*read_html(url)
```
]
 
.panel2-rvest-auto[

```
## {html_document}
## &lt;html lang="es"&gt;
## [1] &lt;head&gt;\n&lt;meta charset="utf-8"&gt;\n&lt;meta name="viewport" content="width=devi ...
## [2] &lt;body&gt;\n\n&lt;noscript&gt;&lt;iframe src="https://www.googletagmanager.com/ns.html ...
```
]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"

read_html(url) |&gt;
* html_table()
```
]
 
.panel2-rvest-auto[

```
## [[1]]
## # A tibble: 26 × 3
##    Fecha    `Empresa y razón de reclamo`                                      ``
##    &lt;chr&gt;    &lt;chr&gt;                                                          &lt;int&gt;
##  1 28-11-22 Pontificia Universidad Católica - Cuenta correo Exalumnos        109
##  2 13-10-22 No Resuelto Pontificia Universidad Católica - Discriminación     168
##  3 06-01-21 Universidad Católica de Chile - No me llegó libro                303
##  4 15-08-19 No Resuelto universidad católica - falta de seguridad            682
##  5 07-06-19 salud                                                            410
##  6 06-06-19 No Resuelto Universidad católica de Chile - Irresponsabilidad…   781
##  7 08-02-18 No Resuelto Salud - No pagó de licencia                          575
##  8 06-01-18 puc - deuda                                                      780
##  9 30-03-17 No Resuelto Pontificia Universidad Católica de Chile - Falta …  1361
## 10 29-12-16 No Resuelto Pontificia Universidad Católica - Crédito UC Desp…  1022
## # … with 16 more rows
```
]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"

read_html(url) |&gt;
  html_table() |&gt;
* first()
```
]
 
.panel2-rvest-auto[

```
## # A tibble: 26 × 3
##    Fecha    `Empresa y razón de reclamo`                                      ``
##    &lt;chr&gt;    &lt;chr&gt;                                                          &lt;int&gt;
##  1 28-11-22 Pontificia Universidad Católica - Cuenta correo Exalumnos        109
##  2 13-10-22 No Resuelto Pontificia Universidad Católica - Discriminación     168
##  3 06-01-21 Universidad Católica de Chile - No me llegó libro                303
##  4 15-08-19 No Resuelto universidad católica - falta de seguridad            682
##  5 07-06-19 salud                                                            410
##  6 06-06-19 No Resuelto Universidad católica de Chile - Irresponsabilidad…   781
##  7 08-02-18 No Resuelto Salud - No pagó de licencia                          575
##  8 06-01-18 puc - deuda                                                      780
##  9 30-03-17 No Resuelto Pontificia Universidad Católica de Chile - Falta …  1361
## 10 29-12-16 No Resuelto Pontificia Universidad Católica - Crédito UC Desp…  1022
## # … with 16 more rows
```
]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"

read_html(url) |&gt;
  html_table() |&gt;
  first() |&gt;
* set_names(c("fecha", "reclamo", "cantidad"))
```
]
 
.panel2-rvest-auto[

```
## # A tibble: 26 × 3
##    fecha    reclamo                                                      canti…¹
##    &lt;chr&gt;    &lt;chr&gt;                                                          &lt;int&gt;
##  1 28-11-22 Pontificia Universidad Católica - Cuenta correo Exalumnos        109
##  2 13-10-22 No Resuelto Pontificia Universidad Católica - Discriminación     168
##  3 06-01-21 Universidad Católica de Chile - No me llegó libro                303
##  4 15-08-19 No Resuelto universidad católica - falta de seguridad            682
##  5 07-06-19 salud                                                            410
##  6 06-06-19 No Resuelto Universidad católica de Chile - Irresponsabilid…     781
##  7 08-02-18 No Resuelto Salud - No pagó de licencia                          575
##  8 06-01-18 puc - deuda                                                      780
##  9 30-03-17 No Resuelto Pontificia Universidad Católica de Chile - Falt…    1361
## 10 29-12-16 No Resuelto Pontificia Universidad Católica - Crédito UC De…    1022
## # … with 16 more rows, and abbreviated variable name ¹​cantidad
```
]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"

read_html(url) |&gt;
  html_table() |&gt;
  first() |&gt;
  set_names(c("fecha", "reclamo", "cantidad")) |&gt;
* mutate(fecha = dmy(fecha))
```
]
 
.panel2-rvest-auto[

```
## # A tibble: 26 × 3
##    fecha      reclamo                                                    canti…¹
##    &lt;date&gt;     &lt;chr&gt;                                                        &lt;int&gt;
##  1 2022-11-28 Pontificia Universidad Católica - Cuenta correo Exalumnos      109
##  2 2022-10-13 No Resuelto Pontificia Universidad Católica - Discriminac…     168
##  3 2021-01-06 Universidad Católica de Chile - No me llegó libro              303
##  4 2019-08-15 No Resuelto universidad católica - falta de seguridad          682
##  5 2019-06-07 salud                                                          410
##  6 2019-06-06 No Resuelto Universidad católica de Chile - Irresponsabil…     781
##  7 2018-02-08 No Resuelto Salud - No pagó de licencia                        575
##  8 2018-01-06 puc - deuda                                                    780
##  9 2017-03-30 No Resuelto Pontificia Universidad Católica de Chile - Fa…    1361
## 10 2016-12-29 No Resuelto Pontificia Universidad Católica - Crédito UC …    1022
## # … with 16 more rows, and abbreviated variable name ¹​cantidad
```
]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"

read_html(url) |&gt;
  html_table() |&gt;
  first() |&gt;
  set_names(c("fecha", "reclamo", "cantidad")) |&gt;
  mutate(fecha = dmy(fecha)) |&gt;
* arrange(fecha, cantidad)
```
]
 
.panel2-rvest-auto[

```
## # A tibble: 26 × 3
##    fecha      reclamo                                                    canti…¹
##    &lt;date&gt;     &lt;chr&gt;                                                        &lt;int&gt;
##  1 2013-05-24 Pontificia Universidad Católica - Sin Respuesta de Reclam…    2677
##  2 2013-07-11 Pontificia Universidad Católica - termino repentino de su…    1569
##  3 2013-10-16 pontificia Universidad Católica (salud) - no respetar con…    2045
##  4 2014-01-02 Pontificia Universidad Católica red salud uc - Discrimina…    2628
##  5 2014-04-07 Salud - Robo                                                  1076
##  6 2014-07-29 salud - demora resolución licencia                             984
##  7 2014-08-10 Pontificia Universidad Católica - Pérdida IRRESPONSABLE d…    1404
##  8 2014-09-16 Pontificia Universidad Católica de Chile - Falta de plata…    1302
##  9 2015-02-18 salud - mala atención                                         1623
## 10 2015-05-12 No Resuelto Pontificia Universidad Católica - Falta de pr…    2739
## # … with 16 more rows, and abbreviated variable name ¹​cantidad
```
]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"

read_html(url) |&gt;
  html_table() |&gt;
  first() |&gt;
  set_names(c("fecha", "reclamo", "cantidad")) |&gt;
  mutate(fecha = dmy(fecha)) |&gt;
  arrange(fecha, cantidad) |&gt;
* ggplot(aes(fecha, cantidad))
```
]
 
.panel2-rvest-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/rvest_auto_10_output-1.svg" width="100%" /&gt;
]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"

read_html(url) |&gt;
  html_table() |&gt;
  first() |&gt;
  set_names(c("fecha", "reclamo", "cantidad")) |&gt;
  mutate(fecha = dmy(fecha)) |&gt;
  arrange(fecha, cantidad) |&gt;
  ggplot(aes(fecha, cantidad)) +
* geom_line(color = "gray80", linewidth = 1.2)
```
]
 
.panel2-rvest-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/rvest_auto_11_output-1.svg" width="100%" /&gt;
]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"

read_html(url) |&gt;
  html_table() |&gt;
  first() |&gt;
  set_names(c("fecha", "reclamo", "cantidad")) |&gt;
  mutate(fecha = dmy(fecha)) |&gt;
  arrange(fecha, cantidad) |&gt;
  ggplot(aes(fecha, cantidad)) +
  geom_line(color = "gray80", linewidth = 1.2) +
* geom_smooth(alpha = 0.25, size = 1.2)
```
]
 
.panel2-rvest-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/rvest_auto_12_output-1.svg" width="100%" /&gt;
]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"

read_html(url) |&gt;
  html_table() |&gt;
  first() |&gt;
  set_names(c("fecha", "reclamo", "cantidad")) |&gt;
  mutate(fecha = dmy(fecha)) |&gt;
  arrange(fecha, cantidad) |&gt;
  ggplot(aes(fecha, cantidad)) +
  geom_line(color = "gray80", linewidth = 1.2) +
  geom_smooth(alpha = 0.25, size = 1.2) +
* coord_cartesian(ylim = c(0, NA))
```
]
 
.panel2-rvest-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/rvest_auto_13_output-1.svg" width="100%" /&gt;
]

---
count: false
 
## HTML &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest-auto[

```r
library(rvest)
library(lubridate)

url &lt;- "https://www.reclamos.cl/empresa/pontificia_universidad_catolica"

read_html(url) |&gt;
  html_table() |&gt;
  first() |&gt;
  set_names(c("fecha", "reclamo", "cantidad")) |&gt;
  mutate(fecha = dmy(fecha)) |&gt;
  arrange(fecha, cantidad) |&gt;
  ggplot(aes(fecha, cantidad)) +
  geom_line(color = "gray80", linewidth = 1.2) +
  geom_smooth(alpha = 0.25, size = 1.2) +
  coord_cartesian(ylim = c(0, NA)) +
* theme_minimal()
```
]
 
.panel2-rvest-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/rvest_auto_14_output-1.svg" width="100%" /&gt;
]

&lt;style&gt;
.panel1-rvest-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel2-rvest-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel3-rvest-auto {
  color: black;
  width: NA%;
  hight: 33%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
&lt;/style&gt;






---

count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
*library(rvest)
```
]
 
.panel2-rvest2-auto[

]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
*library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
```
]
 
.panel2-rvest2-auto[

]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
*library(chilemapas)  # gracias Pachá!
```
]
 
.panel2-rvest2-auto[

]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
*url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"
```
]
 
.panel2-rvest2-auto[

]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"

*read_html(url)
```
]
 
.panel2-rvest2-auto[

```
## {html_document}
## &lt;html lang="en"&gt;
## [1] &lt;head&gt;\n&lt;meta http-equiv="Content-Type" content="text/html; charset=UTF-8 ...
## [2] &lt;body&gt;\n    &lt;div class="main-container"&gt;\n      &lt;header&gt;&lt;div class="inf"&gt; ...
```
]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"

read_html(url) |&gt;
* html_table()
```
]
 
.panel2-rvest2-auto[

```
## [[1]]
## # A tibble: 1 × 6
##   X1    X2    X3    X4    X5    X6   
##   &lt;lgl&gt; &lt;lgl&gt; &lt;lgl&gt; &lt;lgl&gt; &lt;lgl&gt; &lt;lgl&gt;
## 1 NA    NA    NA    NA    NA    NA   
## 
## [[2]]
## # A tibble: 21 × 5
##    `Fecha Local / Lugar`                         Fecha…¹ Latit…² Profu…³ Magni…⁴
##    &lt;chr&gt;                                         &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  
##  1 2023-04-19 17:28:0930 km al E de Concepción   2023-0… -36.94… 33 km   3.0 Ml 
##  2 2023-04-19 17:08:1231 km al SO de Mina Colla… 2023-0… -21.03… 109 km  2.7 Ml 
##  3 2023-04-19 17:02:5373 km al SE de Camiña      2023-0… -19.86… 99 km   2.8 Ml 
##  4 2023-04-19 15:28:1675 km al E de Tierra Amar… 2023-0… -27.58… 107 km  3.3 Ml 
##  5 2023-04-19 14:53:5439 km al O de La Serena    2023-0… -29.82… 30 km   3.0 Ml 
##  6 2023-04-19 14:04:5441 km al S de Curicó       2023-0… -35.33… 103 km  2.5 Ml 
##  7 2023-04-19 13:33:09114 km al O de Tirúa       2023-0… -38.47… 9 km    3.3 Ml 
##  8 2023-04-19 12:00:2582 km al SE de Socaire     2023-0… -24.16… 213 km  2.8 Ml 
##  9 2023-04-19 10:45:33122 km al SE de Socaire    2023-0… -24.33… 175 km  2.8 Ml 
## 10 2023-04-19 08:52:0176 km al NE de Calama      2023-0… -21.91… 108 km  2.7 Ml 
## # … with 11 more rows, and abbreviated variable names ¹​`Fecha UTC`,
## #   ²​`Latitud / Longitud`, ³​Profundidad, ⁴​`Magnitud (2)`
```
]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"

read_html(url) |&gt;
  html_table() |&gt;
* nth(2)
```
]
 
.panel2-rvest2-auto[

```
## # A tibble: 21 × 5
##    `Fecha Local / Lugar`                         Fecha…¹ Latit…² Profu…³ Magni…⁴
##    &lt;chr&gt;                                         &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  
##  1 2023-04-19 17:28:0930 km al E de Concepción   2023-0… -36.94… 33 km   3.0 Ml 
##  2 2023-04-19 17:08:1231 km al SO de Mina Colla… 2023-0… -21.03… 109 km  2.7 Ml 
##  3 2023-04-19 17:02:5373 km al SE de Camiña      2023-0… -19.86… 99 km   2.8 Ml 
##  4 2023-04-19 15:28:1675 km al E de Tierra Amar… 2023-0… -27.58… 107 km  3.3 Ml 
##  5 2023-04-19 14:53:5439 km al O de La Serena    2023-0… -29.82… 30 km   3.0 Ml 
##  6 2023-04-19 14:04:5441 km al S de Curicó       2023-0… -35.33… 103 km  2.5 Ml 
##  7 2023-04-19 13:33:09114 km al O de Tirúa       2023-0… -38.47… 9 km    3.3 Ml 
##  8 2023-04-19 12:00:2582 km al SE de Socaire     2023-0… -24.16… 213 km  2.8 Ml 
##  9 2023-04-19 10:45:33122 km al SE de Socaire    2023-0… -24.33… 175 km  2.8 Ml 
## 10 2023-04-19 08:52:0176 km al NE de Calama      2023-0… -21.91… 108 km  2.7 Ml 
## # … with 11 more rows, and abbreviated variable names ¹​`Fecha UTC`,
## #   ²​`Latitud / Longitud`, ³​Profundidad, ⁴​`Magnitud (2)`
```
]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"

read_html(url) |&gt;
  html_table() |&gt;
  nth(2) |&gt;
* mutate(across(everything(), readr::parse_guess))
```
]
 
.panel2-rvest2-auto[

```
## # A tibble: 21 × 5
##    `Fecha Local / Lugar`             `Fecha UTC`         Latit…¹ Profu…² Magni…³
##    &lt;chr&gt;                             &lt;dttm&gt;              &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  
##  1 2023-04-19 17:28:0930 km al E de… 2023-04-19 21:28:09 -36.94… 33 km   3.0 Ml 
##  2 2023-04-19 17:08:1231 km al SO d… 2023-04-19 21:08:12 -21.03… 109 km  2.7 Ml 
##  3 2023-04-19 17:02:5373 km al SE d… 2023-04-19 21:02:53 -19.86… 99 km   2.8 Ml 
##  4 2023-04-19 15:28:1675 km al E de… 2023-04-19 19:28:16 -27.58… 107 km  3.3 Ml 
##  5 2023-04-19 14:53:5439 km al O de… 2023-04-19 18:53:54 -29.82… 30 km   3.0 Ml 
##  6 2023-04-19 14:04:5441 km al S de… 2023-04-19 18:04:54 -35.33… 103 km  2.5 Ml 
##  7 2023-04-19 13:33:09114 km al O d… 2023-04-19 17:33:09 -38.47… 9 km    3.3 Ml 
##  8 2023-04-19 12:00:2582 km al SE d… 2023-04-19 16:00:25 -24.16… 213 km  2.8 Ml 
##  9 2023-04-19 10:45:33122 km al SE … 2023-04-19 14:45:33 -24.33… 175 km  2.8 Ml 
## 10 2023-04-19 08:52:0176 km al NE d… 2023-04-19 12:52:01 -21.91… 108 km  2.7 Ml 
## # … with 11 more rows, and abbreviated variable names ¹​`Latitud / Longitud`,
## #   ²​Profundidad, ³​`Magnitud (2)`
```
]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"

read_html(url) |&gt;
  html_table() |&gt;
  nth(2) |&gt;
  mutate(across(everything(), readr::parse_guess)) |&gt;
* mutate(Magnitud = as.numeric(stringr::str_remove(`Magnitud (2)`, "Ml")))
```
]
 
.panel2-rvest2-auto[

```
## # A tibble: 21 × 6
##    `Fecha Local / Lugar`     `Fecha UTC`         Latit…¹ Profu…² Magni…³ Magni…⁴
##    &lt;chr&gt;                     &lt;dttm&gt;              &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;     &lt;dbl&gt;
##  1 2023-04-19 17:28:0930 km… 2023-04-19 21:28:09 -36.94… 33 km   3.0 Ml      3  
##  2 2023-04-19 17:08:1231 km… 2023-04-19 21:08:12 -21.03… 109 km  2.7 Ml      2.7
##  3 2023-04-19 17:02:5373 km… 2023-04-19 21:02:53 -19.86… 99 km   2.8 Ml      2.8
##  4 2023-04-19 15:28:1675 km… 2023-04-19 19:28:16 -27.58… 107 km  3.3 Ml      3.3
##  5 2023-04-19 14:53:5439 km… 2023-04-19 18:53:54 -29.82… 30 km   3.0 Ml      3  
##  6 2023-04-19 14:04:5441 km… 2023-04-19 18:04:54 -35.33… 103 km  2.5 Ml      2.5
##  7 2023-04-19 13:33:09114 k… 2023-04-19 17:33:09 -38.47… 9 km    3.3 Ml      3.3
##  8 2023-04-19 12:00:2582 km… 2023-04-19 16:00:25 -24.16… 213 km  2.8 Ml      2.8
##  9 2023-04-19 10:45:33122 k… 2023-04-19 14:45:33 -24.33… 175 km  2.8 Ml      2.8
## 10 2023-04-19 08:52:0176 km… 2023-04-19 12:52:01 -21.91… 108 km  2.7 Ml      2.7
## # … with 11 more rows, and abbreviated variable names ¹​`Latitud / Longitud`,
## #   ²​Profundidad, ³​`Magnitud (2)`, ⁴​Magnitud
```
]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"

read_html(url) |&gt;
  html_table() |&gt;
  nth(2) |&gt;
  mutate(across(everything(), readr::parse_guess)) |&gt;
  mutate(Magnitud = as.numeric(stringr::str_remove(`Magnitud (2)`, "Ml"))) |&gt;
* separate(`Latitud / Longitud`, c("lat", "lon"), sep = " ", convert = TRUE)
```
]
 
.panel2-rvest2-auto[

```
## # A tibble: 21 × 7
##    `Fecha Local / Lugar` `Fecha UTC`           lat   lon Profu…¹ Magni…² Magni…³
##    &lt;chr&gt;                 &lt;dttm&gt;              &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;     &lt;dbl&gt;
##  1 2023-04-19 17:28:093… 2023-04-19 21:28:09 -36.9 -72.7 33 km   3.0 Ml      3  
##  2 2023-04-19 17:08:123… 2023-04-19 21:08:12 -21.0 -68.9 109 km  2.7 Ml      2.7
##  3 2023-04-19 17:02:537… 2023-04-19 21:02:53 -19.9 -69.0 99 km   2.8 Ml      2.8
##  4 2023-04-19 15:28:167… 2023-04-19 19:28:16 -27.6 -69.5 107 km  3.3 Ml      3.3
##  5 2023-04-19 14:53:543… 2023-04-19 18:53:54 -29.8 -71.6 30 km   3.0 Ml      3  
##  6 2023-04-19 14:04:544… 2023-04-19 18:04:54 -35.3 -71.1 103 km  2.5 Ml      2.5
##  7 2023-04-19 13:33:091… 2023-04-19 17:33:09 -38.5 -74.8 9 km    3.3 Ml      3.3
##  8 2023-04-19 12:00:258… 2023-04-19 16:00:25 -24.2 -67.4 213 km  2.8 Ml      2.8
##  9 2023-04-19 10:45:331… 2023-04-19 14:45:33 -24.3 -67.0 175 km  2.8 Ml      2.8
## 10 2023-04-19 08:52:017… 2023-04-19 12:52:01 -21.9 -68.5 108 km  2.7 Ml      2.7
## # … with 11 more rows, and abbreviated variable names ¹​Profundidad,
## #   ²​`Magnitud (2)`, ³​Magnitud
```
]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"

read_html(url) |&gt;
  html_table() |&gt;
  nth(2) |&gt;
  mutate(across(everything(), readr::parse_guess)) |&gt;
  mutate(Magnitud = as.numeric(stringr::str_remove(`Magnitud (2)`, "Ml"))) |&gt;
  separate(`Latitud / Longitud`, c("lat", "lon"), sep = " ", convert = TRUE) |&gt;
* ggplot()
```
]
 
.panel2-rvest2-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/rvest2_auto_11_output-1.svg" width="100%" /&gt;
]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"

read_html(url) |&gt;
  html_table() |&gt;
  nth(2) |&gt;
  mutate(across(everything(), readr::parse_guess)) |&gt;
  mutate(Magnitud = as.numeric(stringr::str_remove(`Magnitud (2)`, "Ml"))) |&gt;
  separate(`Latitud / Longitud`, c("lat", "lon"), sep = " ", convert = TRUE) |&gt;
  ggplot() +
* geom_sf(
*   data = chilemapas::mapa_comunas, aes(geometry = geometry),
*   fill = "gray70", color = "white", alpha = 0.5, size = 0.3
*   )
```
]
 
.panel2-rvest2-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/rvest2_auto_12_output-1.svg" width="100%" /&gt;
]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"

read_html(url) |&gt;
  html_table() |&gt;
  nth(2) |&gt;
  mutate(across(everything(), readr::parse_guess)) |&gt;
  mutate(Magnitud = as.numeric(stringr::str_remove(`Magnitud (2)`, "Ml"))) |&gt;
  separate(`Latitud / Longitud`, c("lat", "lon"), sep = " ", convert = TRUE) |&gt;
  ggplot() +
  geom_sf(
    data = chilemapas::mapa_comunas, aes(geometry = geometry),
    fill = "gray70", color = "white", alpha = 0.5, size = 0.3
    ) +
* geom_point(aes(x = lon, y = lat, size = Magnitud),
*   shape = 21, fill = "gray60", color = "gray40",)
```
]
 
.panel2-rvest2-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/rvest2_auto_13_output-1.svg" width="100%" /&gt;
]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"

read_html(url) |&gt;
  html_table() |&gt;
  nth(2) |&gt;
  mutate(across(everything(), readr::parse_guess)) |&gt;
  mutate(Magnitud = as.numeric(stringr::str_remove(`Magnitud (2)`, "Ml"))) |&gt;
  separate(`Latitud / Longitud`, c("lat", "lon"), sep = " ", convert = TRUE) |&gt;
  ggplot() +
  geom_sf(
    data = chilemapas::mapa_comunas, aes(geometry = geometry),
    fill = "gray70", color = "white", alpha = 0.5, size = 0.3
    ) +
  geom_point(aes(x = lon, y = lat, size = Magnitud),
    shape = 21, fill = "gray60", color = "gray40",) +
* coord_sf(xlim = c(-80, NA))
```
]
 
.panel2-rvest2-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/rvest2_auto_14_output-1.svg" width="100%" /&gt;
]

---
count: false
 
## HTML v2 &lt;code&gt;rvest::read_html&lt;/code&gt;
.panel1-rvest2-auto[

```r
library(rvest)
library(tidyverse) # dplyr + ggplot + etc  # dplyr + ggplot + etc
library(chilemapas)  # gracias Pachá!
url &lt;- "https://www.sismologia.cl/sismicidad/catalogo/2023/04/20230419.html"

read_html(url) |&gt;
  html_table() |&gt;
  nth(2) |&gt;
  mutate(across(everything(), readr::parse_guess)) |&gt;
  mutate(Magnitud = as.numeric(stringr::str_remove(`Magnitud (2)`, "Ml"))) |&gt;
  separate(`Latitud / Longitud`, c("lat", "lon"), sep = " ", convert = TRUE) |&gt;
  ggplot() +
  geom_sf(
    data = chilemapas::mapa_comunas, aes(geometry = geometry),
    fill = "gray70", color = "white", alpha = 0.5, size = 0.3
    ) +
  geom_point(aes(x = lon, y = lat, size = Magnitud),
    shape = 21, fill = "gray60", color = "gray40",) +
  coord_sf(xlim = c(-80, NA)) +
* theme_void()
```
]
 
.panel2-rvest2-auto[
&lt;img src="06-Manipulacion-de-datos-importación-readr-readxl-httr-rvest_files/figure-html/rvest2_auto_15_output-1.svg" width="100%" /&gt;
]

&lt;style&gt;
.panel1-rvest2-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel2-rvest2-auto {
  color: black;
  width: 49%;
  hight: 32%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
.panel3-rvest2-auto {
  color: black;
  width: NA%;
  hight: 33%;
  float: left;
  padding-left: 1%;
  font-size: 80%
}
&lt;/style&gt;





---

## Ejercicio BBDD

Revise https://relational.fit.cvut.cz/dataset/Employee.

1. Realice la conexión `con` a la bbdd.
1. Obtenga el salario promedio por título y genero. 
1. Utilice el comando `show_query` para explorar la consulta.
1. Grafique de alguna forma adecuada los datos obtenidos.

---

## Ejercicio archivos de texto

Lea un archivo de texto utilizando alguna función del paquete `readr` y 
con la funcion `data.table::fread`. Que obtiene con dada una? Existen Diferencas?


---

## Más información sobre transformación de datos

- Paquete [dbplyr](https://dbplyr.tidyverse.org/articles/dbplyr.html).


    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"slideNumberFormat": "%current%",
"highlightStyle": "github",
"highlightLines": true,
"ratio": "16:9",
"countIncrementalSlides": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
